### PR TITLE
feat(23): adding question BE

### PR DESCRIPTION
**Pull request by alexZogich**
## PR Checklist

PR meets the following requirements:

- [x] Changes follow our [guidelines](https://github.com/extreme-startup/questionator/wiki/Contributing)
- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

## PR Type

PR introduces the following changes:

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```
## Actual/New behavior
Its a PR for [FEATURE_23(BE)](https://github.com/extreme-startup/questionator/issues/23)
```
 GET /question  - get all questions
 GET /question/:id - get question by id
 POST /question  - add new question
 PUT /question/:id - update question 
 DELETE /question - delete question

### CODE REVIEW

File: server/src/modules/Question/question.controller.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Controller, Get, Param, Post, Body, Put, Delete } from '@nestjs/common';
+import { QuestionDto } from './dto/question.dto';
+import { QuestionService } from './question.service';
+
+@Controller('question')
```
Comment (kovnick) [inline server/src/modules/Question/question.controller.ts]: Probably 'questions' will be better (plural)
with current realization /question endpoint will return list of all questions

File: server/src/modules/Question/question.controller.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export class QuestionController {
+
+    constructor(private questionService: QuestionService) {
+
+    }
+
+    @Get()
+    async findAll(): Promise<QuestionDto[]> {
+        return await this.findAll();
```
Comment (kovnick) [inline server/src/modules/Question/question.controller.ts]: Should it be this.questionService.findAll()?

File: server/src/modules/Question/question.controller.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Get(':id')
+    async findOneById(@Param() params): Promise<QuestionDto> {
+        return await this.questionService.findById(params.id);
+    }
+
+    @Post()
+    async create(@Body() book: QuestionDto): Promise<QuestionDto> {
+        return await this.questionService.insert(book);
```
Comment (kovnick) [inline server/src/modules/Question/question.controller.ts]: book?

File: server/src/modules/Question/question.controller.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Put(':id')
+    async update(@Body() updatedQuestion: QuestionDto, @Param() params): Promise<QuestionDto> {
+        const oldBook = await this.questionService.findById(params.id);
```
Comment (kovnick) [inline server/src/modules/Question/question.controller.ts]: oldBook?

File: server/src/modules/Question/question.controller.ts (hunk 1 above)
Comment (alexZogich) [reply server/src/modules/Question/question.controller.ts]: And its ok, if we want to get specific one we should use /question/id:

File: server/src/modules/Question/question.controller.ts (hunk 4 above)
Comment (alexZogich) [reply server/src/modules/Question/question.controller.ts]: no more books :)

File: server/src/modules/Question/question.controller.ts (hunk 3 above)
Comment (alexZogich) [reply server/src/modules/Question/question.controller.ts]: no more books :)

File: server/src/modules/Question/question.service.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+export class QuestionService {
+
+    constructor(
+        @InjectRepository(Questions)
+        private readonly questionRepository: Repository<Questions>,
+    ) { }
+
+    async findAll(): Promise<Questions[]> {
+        try {
+            return await this.questionRepository.find({});
```
Comment (MelnychenkoIvan) [inline server/src/modules/Question/question.service.ts]: Is it need to pass an empty object to the find method?

File: server/src/modules/Question/question.service.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        try {
+            return await this.questionRepository.save(updatedQuestion);
+        } catch (err) {
+            return err;
+        }
+
+    }
+
+    async delete(id: string) {
```
Comment (MelnychenkoIvan) [inline server/src/modules/Question/question.service.ts]: As I remember, we discussed that delete method have to do a soft delete and should implemented with low priority, right? So, is that method should be implemented here?

File: server/src/entity/Questions.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({name: 'qms_question'})
+export class Questions {
```
Comment (KITSAnatoliiKarelov) [inline server/src/entity/Questions.ts]: Why plural?

File: server/src/entity/Questions.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    type: string;
+
+    @Column('varchar', {
+        name: 'text',
+        nullable: false,
+    })
+    text: string;
+
+    @Column('varchar', {
+        name: 'answer',
```
Comment (KITSAnatoliiKarelov) [inline server/src/entity/Questions.ts]:
```suggestion
        name: 'correct_answer',
```
As there is going to be a class `AnsweredQuestions` extending this one with a field `competitor_answer` / `obtained_answer`

File: server/src/modules/Question/question.controller.ts (hunk 2 above)
Comment (alexZogich) [reply server/src/modules/Question/question.controller.ts]: yep, fixed

File: server/src/modules/Question/question.service.ts (hunk 5 above)
Comment (alexZogich) [reply server/src/modules/Question/question.service.ts]: for finding all records no need, removed

File: server/src/modules/Question/question.service.ts (hunk 6 above)
Comment (alexZogich) [reply server/src/modules/Question/question.service.ts]: For consistency purposes implemented like [here](https://github.com/extreme-startup/questionator/pull/62)

File: server/src/entity/Questions.ts (hunk 7 above)
Comment (alexZogich) [reply server/src/entity/Questions.ts]: Singular now

**Pull request merged**